examples/foomedical/src/pages/account/Profile.tsx

```diff
@@ -1,23 +1,69 @@
-import { Box, NativeSelect, Stack, TextInput, Title } from '@mantine/core';
-import { formatAddress, formatFamilyName, formatGivenName, formatHumanName } from '@medplum/core';
-import { HumanName, Patient } from '@medplum/fhirtypes';
-import { Form, ResourceAvatar, useMedplumProfile } from '@medplum/react';
+import { Box, Button, InputLabel, LoadingOverlay, NativeSelect, Stack, TextInput, Title } from '@mantine/core';
+import { showNotification } from '@mantine/notifications';
+import { AddressInput, Form, ResourceAvatar, useMedplum } from '@medplum/react';
+import { formatFamilyName, formatGivenName, formatHumanName, normalizeErrorString } from '@medplum/core';
+import { Address, HumanName, Patient } from '@medplum/fhirtypes';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { useState } from 'react';
 import { InfoSection } from '../../components/InfoSection';
+import { cleanResource } from '../../components/utils';
 
 export function Profile(): JSX.Element | null {
-  const profile = useMedplumProfile() as Patient;
+  const medplum = useMedplum();
+  const [profile, setProfile] = useState<Patient>(medplum.getProfile() as Patient);
+  const [loading, setLoading] = useState(false);
+  const [address, setAddress] = useState<Address>(profile.address?.[0] || {});
+
+  async function handleProfileEdit(formData: Record<string, string>): Promise<void> {
+    setLoading(true);
+    const newProfile: Patient = {
+      ...profile,
+      name: [
+        {
+          use: 'official',
+          given: [formData.givenName],
+          family: formData.familyName,
+        },
+      ],
+      birthDate: formData.birthDate,
+      gender: formData.gender as Patient['gender'],
+      address: [address],
+    };
+    const updatedProfile = await medplum
+      .updateResource(cleanResource(newProfile))
+      .then((profile) => {
+        showNotification({
+          icon: <IconCircleCheck />,
+          title: 'Success',
+          message: 'Profile edited',
+        });
+        medplum.getProfile();
+        window.scrollTo(0, 0);
+        return profile;
+      })
```

Comment:
Is there an SDK mechanism I can use to force a profile refresh in the `MedplumContext`? I couldn't use the `refreshProfile` because it is a private method. I tried the `invalidate*` methods but they didn't work.

CC @rahul1